src/script/script.cpp

```diff
@@ -276,7 +276,7 @@ unsigned int CScript::GetSigOpCount(bool fAccurate) const
         opcodetype opcode;
         if (!GetOp(pc, opcode))
             break;
-        if (opcode == OP_CHECKSIG || opcode == OP_CHECKSIGVERIFY)
+        if (opcode == OP_CHECKSIG || opcode == OP_CHECKSIGVERIFY || opcode == OP_DATASIGVERIFY)
```

Comment:
There is a unit test for SigOpCount in `sigopcount_tests.cpp`

Reply:
I considered this, but I decided not to do it because if OP_DATASIGVERIFY is not enabled, the transaction is invalid anyway so the number of sigops inside it is irrelevant.  I wanted to minimize the use of enableDataSigVerify...

Reply:
OK, I hadn't considered that. 

Still, I think you should add the check, because if you don't you're making the correctness of the function depend on how it's used. A subtle gotcha.